### ISSUE TITLE
paho.mqtt.publish.multiple publishes messages in wrong order

### ISSUE THREAD

**Issue by stmax82** [eclipse/paho.mqtt.python#321]
The following code uses publish.multiple to publish multiple messages.

For some reason my subscribers receive the messages in reverse order.

When I send messages with payloads "msg0", "msg1", "msg2", the subscribers receive "msg2", "msg1", "msg0".

Is this normal and as it should be? (Documentation doesn't seem to mention this behavior anywhere...)

Is it safe to reverse the order of messages before sending them?

```python
import paho.mqtt.publish as publish

msgs = [{"topic": "pahotest/foo", "payload": "msg%d" % i, "qos": 2} for i in range(10)]

publish.multiple(msgs, hostname="test.mosquitto.org", port=1883)
```

paho version 1.3.1

**Comment by tarunw07** [comment]
Yes, it is safe to reverse the order of the messages before sending them.

The reason it is publishing in reverse order is that it uses a pop function(which returns the last element of the list) to get the message and then publishes it. I don't know if there will be any change in the next versions so be careful about that(most probably there will be no change).

**Comment by ralight** [comment]
I'd say this is a bug and needs fixing.

**Comment by stmax82** [comment]
@ralight do you have an idea in mind how it should be fixed? Does it have to be backward compatible? Or could we just set userdata=list(reversed(msgs)) in publish.multiple?

```python
    client = paho.Client(client_id=client_id,
                         userdata=list(reversed(msgs)), protocol=protocol, transport=transport)
                                  ^^^^^^^^^^^^^^^^^^^^
```

**Comment by tarunw07** [comment]
A more efficient way would be to reverse the messages using list.reverse() function.

    msgs.reverse()
    client = paho.Client(....)

**Comment by PierreF** [comment]
This bug was fixed 1 year ago by #215... but on develop which is not yet released :(

But to be clear: yes it's a bug, the publish.multiple should send message in order. It will be fixed in next release (should be done today).

**Comment by stmax82** [comment]
Seems to be fixed. I just had a bug report because our MQTT client sends messages in reverse order when used with the latest paho-mqtt version :p

Thanks for the fix!
